Sources/NIOCore/EventLoopFuture.swift

```diff
@@ -1073,6 +1073,26 @@ extension EventLoopFuture {
     @preconcurrency
     @inlinable
     public func wait(file: StaticString = #file, line: UInt = #line) throws -> Value where Value: Sendable {
+        #if os(WASI)
+        // NOTE: As of July 22, 2025 `wait()` calling wait() is not supported on WASI platforms.
+        //
+        // This may change down the road if and when true multi-threading evolves. But right now
+        // calling wait here results in the following runtime crash:
+        //
+        // ```
+        // SomeExecutable.wasm:0x123456 Uncaught (in promise) RuntimeError: Atomics.wait cannot be called in this context
```

Comment:
Does this also happen with multi-threaded Swift SDKs for WASI? If not, we should only make this conditional on multi-threading availability.

Reply:
It's not an SDK limitation, but the browser's limitation. They don't allow exeucting `memory.atomic.wait32` instruction on the main thread regardless of the SDK variants. On multi-threading context, the opt-in busy-wait mode in wasi-libc (https://github.com/WebAssembly/wasi-libc/pull/562) might make sense under some conditions, though.

Reply:
Correct, this is a limitation only applied to the main thread.